src/app/fdctl/run/tiles/metrics.c

```diff
@@ -0,0 +1,392 @@
+               const ulong post_buffer_sz,
+               char * const data ) {
+  int ssl_sockfd = socket( AF_INET, SOCK_STREAM, 0 );
+  if( FD_UNLIKELY( ssl_sockfd < 0 ) ) {
+    FD_LOG_WARNING(( "socket error %d (%d) %s", ssl_sockfd, errno, strerror( errno ) ));
+    return;
+  }
+
+  /* Resolve the server address */
+  struct hostent * server = gethostbyname( _server );
+  if( FD_UNLIKELY( !server ) ) {
+    FD_LOG_WARNING(( "gethostbyname error %d h_errno=%d", ssl_sockfd, h_errno ));
+    goto cleanup0;
+  }
+  struct sockaddr_in server_addr;
+  memset( &server_addr, 0, sizeof(server_addr) );
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_port = htons( port );
+  memcpy( &server_addr.sin_addr.s_addr, server->h_addr, ( size_t )server->h_length );
+
+  /* Connect to the server */
+  struct sockaddr sockaddr;
+  memcpy( &sockaddr, &server_addr, sizeof( struct sockaddr ) );
+  int ret = connect( ssl_sockfd, &sockaddr, sizeof( server_addr ) );
+  if( FD_UNLIKELY( ret < 0 ) ) {
+    FD_LOG_WARNING(( "connect error %d (%d) %s", ret, errno, strerror( errno ) ));
+    goto cleanup0;
+  }
+
+  /* Create an SSL context */
+  SSL_CTX * ctx = SSL_CTX_new( TLS_client_method() );
+  SSL     * ssl = SSL_new( ctx );
+  SSL_set_fd( ssl, ssl_sockfd );
+  SSL_connect( ssl );
+  ulong err = ERR_get_error();
+  if( FD_UNLIKELY( err ) ) {
+    FD_LOG_WARNING(( "OpenSSL error: %s", ERR_error_string( err, NULL ) ));
+    goto cleanup1;
+  }
+
+  snprintf( post_buffer, post_buffer_sz,
+           "POST /write?db=%s HTTP/1.1\r\n"
+           "Host: %s:%d\r\n"
+           "Authorization: Basic %s\r\n"
+           "Content-Length: %zu\r\n"
+           "Content-Type: application/x-www-form-urlencoded\r\n\r\n"
+           "%s", db, _server, port, user_pass, strlen(data), data);
+  if( FD_UNLIKELY( ret<0 || ret>=(int)post_buffer_sz ) ) {
+    FD_LOG_WARNING(( "snprintf error %d", ret ));
+    goto cleanup1;
+  }
+
+  if( DEBUG ) FD_LOG_NOTICE(( "post data %s", post_buffer ));
+
+  ret = SSL_write( ssl, post_buffer, (int)strlen( post_buffer ) );
+  if( FD_UNLIKELY( ret <= 0 ) ) {
+    FD_LOG_WARNING(( "OpenSSL error: %d", SSL_get_error( ssl, ret ) ));
+    goto cleanup1;
+  }
+
+  /* Receive and print the response */
+  char response[4096];
+  ret = SSL_read( ssl, response, sizeof( response ) );
+  if( FD_UNLIKELY( ret <= 0 ) ) {
+    FD_LOG_WARNING(( "OpenSSL error: %d", SSL_get_error( ssl, ret ) ));
+    goto cleanup1;
+  }
+  if( ret > 0 ) {
+    response[ ret ] = '\0';
+    FD_LOG_NOTICE(( "response %s", response ));
+  }
+
+  /* Clean up */
+  cleanup1:
+  SSL_free( ssl );
+  SSL_CTX_free( ctx );
+  cleanup0:
+  close( ssl_sockfd );
+}
+
+static void
+init( fd_tile_args_t * args ) {
+  (void)args;
+  /* calling fd_tempo_tick_per_ns requires nanosleep, it is cached with
+     a FD_ONCE */
+  fd_tempo_tick_per_ns( NULL );
+
+  ssl_init();
+  if( DEBUG ) FD_LOG_NOTICE(( "metrics init" ));
+}
+
+static void
+run( fd_tile_args_t * args ) {
```

Comment:
Could we make this tile use mux like the other ones, and align it to the same style / semantics

Reply:
The metrics tile won't use the mux tile for now since it does not require flow control and it's already implemented.